static/tps/2017_2/tp3/index.md

```diff
@@ -0,0 +1,213 @@
+Bacon Kevin (II),Behind the Scene (2011)
+Bacon Les,Barack Obama on 60 Minutes Talks Osama Bin Laden (2011)
+Bacon Lewis,Beverley (2015)
+```
+
+Este archivo se va a usar para generar un grafo donde los vértices sean actores y las aristas sean las películas en las que hayan colaborado, así conectándolos. 
+
+³Las bases de datos de IMDB eran un poco más confusas que las proporcionadas. Si quieren ver cómo se trabajo con el archivo original ir [acá.](https://github.com/FdelMazo/IMDBtoCSV/blob/master/imdbtocsv.py)
+
+⁴Siendo que `actors.csv` es muy pesado, un archivo más ligero, `test.csv`, también es proporcionado para no tener que procesar el dataset entero a la hora de probar resultados localmente. Este contiene lineas del archivo original escogidas al azar, más la linea de Kevin Bacon.
+
+### Primitivas del grafo
+
+1. Crear grafo (y de ser necesario destruirlo).
+1. Agregar y borrar arista.
+1. Agregar y borrar vértice.
+1. Obtener adyacentes de un vértice.
+1. Verificar si dos vértices son adyacentes.
+1. Obtener aristas.
+1. Obtener vértices.
+
+### Funciones a implementar
+
+1. _Camino_: 
+ 
+   Dado dos actores, devuelve el camino **más corto** que recorre el primer actor para llegar al segundo.
+   ```
+   camino("Clint Eastwood", "Christian Bale")
+   >>> [("Empire of The Sun (1987)", "Clint Eastwood", "Ben Stiller"), ("Unite for Japan (2011)", "Ben Stiller", "Christian Bale")]
+   ```
+
+2. _Distancia_: 
+
+   Devuelve la lista de actores a n pasos del actor (ni más ni menos). De n ser 0 debe devolver una lista con solo el nombre del actor
+   ```
+   distancia("Clint Eastwood", 2)
+   >>> ["Paul Rudd", "Michael Douglas", ...]
+   ```
+
+3. _Cantidad Alcanzados_:
+
+   Devuelve la cantidad de personas a las que puede llegar el actor ingresado. 
+   ```
+   cantidad_alcanzados("Clint Eastwood")
+   >>> 2331685
+   ```
+   
+4. _Popularidad_: 
+ 
+    La popularidad⁵ de un actor puede ser medida por la sumatoria de a cuánta gente alcanza toda persona con la que el actor trabajó alguna vez, multiplicado por la cantidad de películas hechas por él.    
+    ```
+    popularidad("Clint Eastwood")
+    >>>  ???
+    ```
```

Comment:
Lo que se menciona antes de la popularidad de una película dónde queda? Es como que hay dos sets de cosas a implementar y no termina de quedar claro qué es lo que debe hacerse

Reply:
Sobre cómo se calcula, creo que puede terminar beneficiando mucho a los actores que actúen en muchas películas aunque sean re falopa (Sry Nick Cage, but it's true).. aunque tampoco es que sea tanto problema